### PR TITLE
VQVAE Implementation for ADNI Dataset

### PR DESCRIPTION
**Pull request by indiradrr**
This pull request introduces the Vector Quantized Variational Autoencoder (VQ-VAE) module for topic recognition in the PatternAnalysis repository for the ADNI dataset (Task 8) implemented in PyTorch.

It contains the following files:
README.md - report consisting of the overview of the model and the results of training
assets - directory for images to visualize algorithm and results
dataset.py - script to load the ADNI dataset
modules.py - script for model architecture classes
predict.py - script for generating new images and reconstructions from loading a saved model
train.py - script to train the VQVAE model

To use, install all the needed dependencies and add the necessary paths to load the dataset.
I would appreciate the feedback :)

### CODE REVIEW

**Comment by SiyuLiu0329** [comment]
**This is an initial inspection, no action is required at this point**

Difficulty: Hard

Readme Overall:  Good
- Project Overview: Good
- Model: Need more details. Did not mention VQ prior
- Data + Preprocessing: Good
- Usage: missing
- Training / Loss Curve(s): Good
- Result Demonstration: Recon ok, image generation not quite working
- References: Good

Functionality:
- Reconstruction: Good
- Generation: Not working

Code:
- Consistent with Results: Yes
- File Structure: Good
- Commenting: Good
- Commit frequency: Good
- Commit messages: Good

Other Comments:
- Please remove .DS_Store
- Usually the prior should be a learnt neural net, histogram is unlikely to work.

**Comment by shakes76** [comment]
# Marking
## Good Practice (Design/Commenting, TF/Torch Usage)
Adequate design and implementation
Good spacing and comments
Header blocks missing -1
## Recognition Problem 
Solves problem 
Driver Script present
File structure present
Shows Usage & Demo & Visualisation & Data usage, no generation -2
Module present
Commenting
No Data leakage
Difficulty: Hard
## Commit Log
Meaningful commit messages
Progressive commits used
## Documentation
ReadMe acceptable, no usage -1, no refs -1
Model/technical explanation
Good Description and Comments 
Markdown used and PDF submitted
## Pull Request
Successful Pull Request (Working Algorithm Delivered on Time in Correct Branch) 
Feedback required, remove DSTore files -2
Request Description good

**Comment by shakes76** [comment]
Feedback marks possible +2 if the requested changes are made (see above).

**Comment by wangzhaomxy** [comment]
No feedback attempt and no feedback marks granted.

**Comment by indiradrr** [comment]
> Feedback marks possible +2 if the requested changes are made (see above).

I fixed the ReadMe file and removed the .DS_Store file as per the feedback. In case other staff sees this, I've explained on Ed post #510 why the commits are from my friend's account because my laptop is currently broken and I wasn't diligent when commiting and pushing the changes.

**Comment by wangzhaomxy** [comment]
Feedback attempt and feedback marks granted +2.
